### PR TITLE
Unity mono 4.8.0 branch bdwgc submodule

### PR DESCRIPTION
**Pull request by mderoy**
Build mono using unity's BDWGC for garbage collection

### CODE REVIEW

**Comment by piconid** [comment]
@joncham    We don't have to do anything for Android for Krait do we?

File: mono/metadata/boehm-gc.c [hunk 1]
```diff
@@ -226,7 +228,9 @@ mono_gc_base_init (void)
 				}
 				continue;
 			} else if (g_str_has_prefix (opt, "toggleref-test")) {
+#if !HAVE_BDWGC_GC
 				register_test_toggleref_callback ();
```
Comment (joncham) [inline mono/metadata/boehm-gc.c]: Does Mono do anything for this with Boehm?

File: mono/metadata/boehm-gc.c [hunk 2]
```diff
@@ -1545,8 +1556,9 @@ mono_gc_register_finalizer_callbacks (MonoGCFinalizerCallbacks *callbacks)
 		g_error ("Invalid finalizer callback version. Expected %d but got %d\n", MONO_GC_FINALIZER_EXTENSION_VERSION, callbacks->version);
 
 	fin_callbacks = *callbacks;
-
+#if !HAVE_BDWGC_GC
 	GC_set_await_finalize_proc ((void (*) (GC_PTR))fin_notifier);
```
Comment (joncham) [inline mono/metadata/boehm-gc.c]: It looks like Mono implemented this in old Boehm. I am not sure what it does though ATM.

File: mono/metadata/boehm-gc.c (hunk 1 above)
Comment (mderoy) [inline mono/metadata/boehm-gc.c]: Only use appears to do anything with boehm when the MONO_GC_PARAMS environment variable includes (g_str_has_prefix (opt, "toggleref-test")) {

File: mono/metadata/boehm-gc.c (hunk 2 above)
Comment (mderoy) [inline mono/metadata/boehm-gc.c]: The function mono_gc_register_finalizer_callbacks doesn't appear to be used anywhere in mono...just exposed via mono.def

**Pull request merged**